### PR TITLE
Covert to ESM module & bundle improvements

### PR DESCRIPTION
**Pull request by slavik-chapelskyi**
Also: 
- replace lodash & querystring by using URL APIs
- build package with tsc instead of babel
- replace babel with ts-jest for test to run it faster and have test type-checking out of the box
- introduce size-limit to track bundle size. (also added as separate ci step)

### CODE REVIEW

**Review by pihorb — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -1,19 +1,9 @@
 {
   "extends": "@shelf/tsconfig/frontend",
   "compilerOptions": {
-    "declaration": true,
-    "target": "esnext",
-    "strict": true,
     "noEmit": false,
+    "declaration": true,
```
Comment (kchlonShelf) [inline tsconfig.json:5]: Maybe we can also add "[declarationMap](https://www.typescriptlang.org/tsconfig#declarationMap)" property which will improve dev experience by referencing directly to source code

**Review by kchlonShelf — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (slavik-chapelskyi) [reply tsconfig.json:5]: Haven't experience using this property before
Would check this out & consider as separate PR. Have you used this before anywhere? Interesting to see what exactly it generates and how could we use it

Comment (kchlonShelf) [reply tsconfig.json:5]: https://github.com/shelfio/get-video-id/assets/106971064/0b4adbf5-60c7-4b28-861d-3c72c61a5f1e


https://github.com/shelfio/get-video-id/assets/106971064/f8aa70bf-e897-48d5-98bb-4594f837807c

Here you can see the differences of references in builded package, so it targets source files from definitions, however I'd like to check how will it work when is used in some external application. I haven't tried this yet but I've been looking into declarations recently and found this option, hence I suggested it ;)

Comment (slavik-chapelskyi) [reply tsconfig.json:5]: Looks interesting. Worth checking when installed in app, if it resolved to the source code properly

**Pull request merged**